### DISCUSSION TITLE
How to Display Dynamic Images

### DISCUSSION THREAD

**Discussion by 935970041** [eclipse-birt/birt#1637] [Q&A] [answered]
Now the data transferred is url or base64 image data. (It's different every time, so it's a dynamic parameter.) How to configure the birt to parse this type of data and display images properly? It works fine when exporting to pdf.  thanks~

**Comment by speckyspooky** [comment, marked as answer]
You have different way to create the report. The important thing is that the main question is which type of "URL" you will use and in which way the image is represented as base64.
Because the URL can include directly the image or is an online reference. A URL can also include a base64-image.
And if you say that the image is a base64-coded image then it is important is the content of the base64-stream really only a image content?

For the report you can use an standard image object and you need a little bit code to handle the images.
You have to check if it is a URL then you can use directly the URL but if you have a raw image coded as base64 you have to decode the base64 and then you can display the result.

A small example of it is:

```
if ( params["paramImageMap"].toString().indexOf("https://") >=0 ) {
  /* image like URL*/
  this.data = params["paramImageMap"].value
} else {
  /* image like Base64, which must be decoded */
  importPackage(Packages.org.apache.commons.codec.binary);
  this.data=Base64.decodeBase64(params["paramImageMap"].value);
}
```

**Type of possible URL-images:**
![grafik](https://github.com/eclipse-birt/birt/assets/41593722/8e2880c6-600e-4ae8-b128-f850a4d4814d)

**Example of base64-image:**
![grafik](https://github.com/eclipse-birt/birt/assets/41593722/3e6d1479-c3ba-4c8c-b174-3a7aeee4861b)

If you can provide may be an example of each type URL & base64 we can analyse which kind of URL and base64 are given.

**Comment by 935970041** [comment]
Thank you, according to the solution you said to pass the base64 type image, it can display the column normally.
